lib/std/js/jsasynchttpclient.nim

```diff
@@ -0,0 +1,93 @@
+##
+## .. Note:: jsasynchttpclient module requires `-d:nimExperimentalJsfetch`
+when not defined(js):
+  {.fatal: "Module jsasynchttpclient is designed to be used with the JavaScript backend.".}
+
+import std/[jsfetch, asyncjs]
+from std/uri import Uri
+
+type JsAsyncHttpClient* = ref object of JsRoot
+
+func newJsAsyncHttpClient*(): JsAsyncHttpClient = discard
+
+func fetchOptionsImpl(body: cstring; metod: static[cstring]): FetchOptions =
+  unsafeNewFetchOptions(metod = metod, body = body, mode = "cors".cstring, credentials = "include".cstring,
+    cache = "default".cstring, referrerPolicy = "unsafe-url".cstring, keepalive = false)
+
+proc getContent*(self: JsAsyncHttpClient; url: Uri | string): Future[cstring] {.async.} =
+  text(await fetch(cstring($url)))
+
+proc deleteContent*(self: JsAsyncHttpClient; url: Uri | string): Future[cstring] {.async.} =
+  text(await fetch(cstring($url), fetchOptionsImpl("".cstring, "DELETE".cstring)))
+
+proc postContent*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[cstring] {.async.} =
+  text(await fetch(cstring($url), fetchOptionsImpl(body.cstring, "POST".cstring)))
+
+proc putContent*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[cstring] {.async.} =
+  text(await fetch(cstring($url), fetchOptionsImpl(body.cstring, "PUT".cstring)))
+
+proc patchContent*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[cstring] {.async.} =
+  text(await fetch(cstring($url), fetchOptionsImpl(body.cstring, "PATCH".cstring)))
+
+proc get*(self: JsAsyncHttpClient; url: Uri | string): Future[Response] {.async.} =
+  fetch(cstring($url))
+
+proc delete*(self: JsAsyncHttpClient; url: Uri | string): Future[Response] {.async.} =
+  fetch(cstring($url), fetchOptionsImpl("".cstring, "DELETE".cstring))
+
+proc post*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[Response] {.async.} =
+  fetch(cstring($url), fetchOptionsImpl(body.cstring, "POST".cstring))
+
+proc put*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[Response] {.async.} =
+  fetch(cstring($url), fetchOptionsImpl(body.cstring, "PUT".cstring))
+
+proc patch*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[Response] {.async.} =
+  fetch(cstring($url), fetchOptionsImpl(body.cstring, "PATCH".cstring))
+
+proc head*(self: JsAsyncHttpClient; url: Uri | string): Future[Response] {.async.} =
+  fetch(cstring($url), fetchOptionsImpl("".cstring, "HEAD".cstring))
+
+
+runnableExamples("-d:nimExperimentalJsfetch -r:off"):
+  import std/asyncjs
+  from std/jsfetch import Response
+  from std/uri import parseUri, Uri
+
+  let client: JsAsyncHttpClient = newJsAsyncHttpClient()
+  const data: string = """{"key": "value"}"""
+
+  block:
+    proc example() {.async.} =
+      let url: Uri = parseUri("http://nim-lang.org")
+      let content: cstring = await client.getContent(url)
+      let response: Response = await client.get(url)
+    discard example()
```

Comment:
put all these blocks inside a proc so you can have a single `discard` that calls everything:
```nim
proc main {.async.} =
  block:
    let url: Uri = parseUri("http://nim-lang.org")
    let content: cstring = await client.getContent(url)
    let response: Response = await client.get(url)
  block:
    let url: Uri = parseUri("http://httpbin.org/delete")
    let content: cstring = await client.deleteContent(url)
    let response: Response = await client.delete(url)
 ...

discard main()
```